src/e2e/java/teammates/e2e/cases/e2e/FeedbackTextQuestionE2ETest.java

```diff
@@ -0,0 +1,134 @@
+        // add new question exactly like loaded question
+        loadedQuestion.setQuestionNumber(2);
+        feedbackEditPage.addTextQuestion(loadedQuestion);
+
+        feedbackEditPage.verifyTextQuestionDetails(2, questionDetails);
+        verifyPresentInDatastore(loadedQuestion);
+
+        ______TS("copy question");
+        FeedbackQuestionAttributes copiedQuestion = testData.feedbackQuestions.get("qn1ForSecondSession");
+        questionDetails = (FeedbackTextQuestionDetails) copiedQuestion.getQuestionDetails();
+        feedbackEditPage.copyQuestion(copiedQuestion.getCourseId(),
+                copiedQuestion.getQuestionDetails().getQuestionText());
+        copiedQuestion.courseId = course.getId();
+        copiedQuestion.feedbackSessionName = feedbackSession.getFeedbackSessionName();
+        copiedQuestion.setQuestionNumber(3);
+
+        feedbackEditPage.verifyTextQuestionDetails(3, questionDetails);
+        verifyPresentInDatastore(copiedQuestion);
+
+        ______TS("edit question");
+        questionDetails.setRecommendedLength(200);
+        copiedQuestion.questionDetails = questionDetails;
+        feedbackEditPage.editTextQuestion(3, questionDetails);
+
+        feedbackEditPage.verifyTextQuestionDetails(3, questionDetails);
+        verifyPresentInDatastore(copiedQuestion);
+    }
+
+    private void testSubmitPage() {
+        AppUrl url = createUrl(Const.WebPageURIs.SESSION_SUBMISSION_PAGE)
+                .withUserId(student.googleId)
+                .withCourseId(student.course)
+                .withSessionName(feedbackSession.getFeedbackSessionName())
+                .withRegistrationKey(StringHelper.encrypt(getStudent(student).key));
+        FeedbackSubmitPage feedbackSubmitPage = loginAdminToPage(url, FeedbackSubmitPage.class);
+        feedbackSubmitPage.waitForPageToLoad();
+
+        ______TS("verify loaded question");
+        FeedbackQuestionAttributes question = testData.feedbackQuestions.get("qn1ForFirstSession");
+        InstructorAttributes receiver = testData.instructors.get("instructor");
+        question.setQuestionNumber(1);
+        feedbackSubmitPage.verifyTextQuestion(1, (FeedbackTextQuestionDetails) question.getQuestionDetails());
+
+        ______TS("submit response");
+        String questionId = getFeedbackQuestion(question).getId();
+        FeedbackResponseAttributes response = getResponse(questionId, receiver, "<p>This is the response for qn 1</p>");
+        feedbackSubmitPage.submitTextResponse(1, receiver.getName(), response);
+
+        verifyPresentInDatastore(response);
+
+        ______TS("check previous response");
+        feedbackSubmitPage = AppPage.getNewPageInstance(browser, url, FeedbackSubmitPage.class);
+        feedbackSubmitPage.waitForPageToLoad();
+        feedbackSubmitPage.verifyTextResponse(1, receiver.getName(), response);
+
+        ______TS("edit response");
+        String editedResponse = "<p><strong>Edited response</strong></p>";
```

Comment:
Minor suggestion: Shall we make this string longer to see if the response length field can work? Maybe a string that has 100 words?

Reply:
Actually above there is a string of length 7 that aims to test the counting functionality of the response length field. I believe increasing it to 100 should not make a difference unless we are testing for handling of excess number of words,  which I think should not be done for E2E test.